core/transaction-pool/graph/src/pool.rs

```diff
@@ -165,6 +172,36 @@ impl<B: ChainApi> Pool<B> {
 		)
 	}
 
+	/// Revalidate all ready transactions.
+	///
+	/// Returns future that performs validation of all ready transactions and
+	/// then resubmits all transactions back to the pool.
+	pub fn revalidate_ready(&self, at: &BlockId<B::Block>) -> impl Future<Output=Result<(), B::Error>> {
+		let validated_pool = self.validated_pool.clone();
+		let ready = self.validated_pool.ready().map(|tx| tx.data.clone());
```

Comment:
Nope - this only clones encoded runtime extrinsic (without validation results - requires/provides/...). I do not think that `drain()` is a good idea, because revalidation could take some time and (if it, hopefully, ends successfully) the light node won't propagate transactions when revalidation is in progress. Also, drain will return `Arc<Transaction>` (as I have told above) => we'll need this clone anyway :/

Reply:
Ahh, and revalidation could fail because e.g. timeout, missing peers, ... And we would need to reinsert tx with previous validation results if there would be `drain()`. Now we just emit warning in this case.